src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetInfoService.cs

```diff
@@ -250,6 +260,8 @@ private static void ConvertToStringAndFree(ref IntPtr ptr, ref string str)
             }
         }
 
+        public void Dispose() => _asyncTasks.Join();
```

Comment:
So looking at the docs, is this correct? My worry being this remark in Join:

> Calling this method when the caller is not executing within a JoinableTask safely no-ops.

I'd be curious which branch this takes:

https://github.com/microsoft/vs-threading/blob/8f1ef716997c51e2eb690e3fe9165da55662e32d/src/Microsoft.VisualStudio.Threading/JoinableTaskCollection.cs#L124-L134

Mind you, if this _doesn't_ work then I have other code to fix as well. :smile:

Reply:
Yep, looks like this pattern is broken. :-/